core/storage/suggestion/gae_models.py

```diff
@@ -891,17 +894,29 @@ def get_in_review_question_suggestions_by_offset(
                     one of the supplied language codes.
                 next_offset: int. The input offset + the number of results
                     returned by the current query.
+
+        Raises:
+            RuntimeError. If skill_ids is empty.
         """
 
+        filters = [
+            cls.status == STATUS_IN_REVIEW,
+            cls.suggestion_type == feconf.SUGGESTION_TYPE_ADD_QUESTION,
+        ]
+
+        if skill_ids is not None:
+            # If this is not filtered here, gae throws BadQueryError.
+            if len(skill_ids) == 0:
+                raise RuntimeError('skill_ids list can\'t be empty')
+
+            filters.append(cls.target_id.IN(skill_ids))
+
         if sort_key == constants.SUGGESTIONS_SORT_KEY_DATE:
             # The first sort property must be the same as the property to which
             # an inequality filter is applied. Thus, the inequality filter on
             # author_id can not be used here.
             suggestion_query = cls.get_all().filter(
-                datastore_services.all_of(
-                    cls.status == STATUS_IN_REVIEW,
-                    cls.suggestion_type == feconf.SUGGESTION_TYPE_ADD_QUESTION,
-            )).order(-cls.created_on)
+                datastore_services.all_of(*filters)).order(-cls.created_on)
```

Comment:
@nikitaevg Just a note -- in some cases, a topic can have up to 40 skills. Does the query here still work fine in such cases, and if so, how long does it take to run?

Reply:
I don't think it will slow down the query much, given the skill_id field is [indexed](https://github.com/nikitaevg/oppia/blob/3fe53120b2cf09fd716ed81f2e9f61f729d854e6/core/storage/suggestion/gae_models.py#L195).

Also, a similar thing works for translations opportunities - they are [filtered](https://github.com/oppia/oppia/blob/fbb52dafadf14ad7833457456ce3acabf46aa9a2/core/storage/suggestion/gae_models.py#L770) by exploration ids which are constructed for a topic, passed to backend [here](https://github.com/oppia/oppia/blob/fbb52dafadf14ad7833457456ce3acabf46aa9a2/core/controllers/contributor_dashboard.py#L390). I guess there could be even more explorations for a topic than skills, right?

Regarding working or not, I don't see why it wouldn't work if it works for two elements. After all, it's a simple `IN` filtering ([here](https://github.com/oppia/oppia/pull/19146/files#diff-01f44d7b1b2d1be700ecfd1115d1dc9381f52404c1a334b0426e00e2f983d303R912)).

Reply:
Oh wow, why is it implemented that way.

It might actually be broken, you were right, I should've checked that. Based on [this](https://cloud.google.com/datastore/docs/concepts/queries#in) it will work only for 30 elements top if I read this correctly. Do you think we should revert the PR?

Reply:
Ah, good catch on the limit of the IN, I didn't know that there was a max 30. I wonder what it does when there are > 30 clauses...does it error silently or throw an exception or something else. Hmmm.

Anyway, I think it's fine to fix-forward this in https://github.com/oppia/oppia/pull/19221. Thanks!